tests/integration-tests/tests/storage/kms_key_factory.py

```diff
@@ -0,0 +1,207 @@
+import json
+import logging
+import random
+import string
+import time
+
+import boto3
+import pkg_resources
+from jinja2 import Environment, FileSystemLoader
+
+
+class KMSKeyFactory:
```

Comment:
From our discussion offline, I think the goal is to use 1 key and 1 role for all tests requiring kms key, the scope of the fixture or how you implement it is entirely up to you

Reply:
Done!